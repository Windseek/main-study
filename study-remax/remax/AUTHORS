Cai Huanyu <eterlf41@gmail.com>
Chase Wu <chase.wu@vitualreaty.com>
ColMugX <colmugx@qq.com>
Meck Zhu <yesmeck@gmail.com>
Pochodaydayup <13343460801@163.com>
QC-L <github@liqichang.com>
Richard Jiang <ahonn95@outlook.com>
Wei Zhu <yesmeck@gmail.com>
ahonn <ahonn95@outlook.com>
caihuanyu <eterlf41@gmail.com>
insane <luojieyy@gmail.com>
jinxin.sjx <jinxin.sjx@alibaba-inc.com>
当轩 <code.falling@gmail.com>
当轩 <me@dangxuan.pro>
愚道 <tingzhao.ytz@antfin.com>
陈帅 <qixian.cs@outlook.com>
