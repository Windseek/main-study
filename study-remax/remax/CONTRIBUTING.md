# 贡献指南

这篇指南会指导你如何为 Remax 贡献一份自己的力量，请在你要提 issue 或者 pull request 之前花几分钟来阅读一遍这篇指南。

## 参与者公约

我们有一份社区的[《参与者公约》](/CONTRIBUTING.md)，希望所有的参与者都能遵守，请花时间阅读一遍全文以确保你能明白哪些是可以做的，哪些是不可以做的。

## 透明的开发

我们所有的工作都会放在 GitHub 上。不管是核心团队的成员还是外部贡献者的 pull request 都需要经过同样流程的 review。

## 分支管理

目前我们所有的开发都在 `master` 分支上进行，我们会尽力保证 `master` 上的代码稳定。所有在 `master` 上提交的代码必须是向下兼容的，可以新增功能，但不允许有破坏性的修改。

## Bugs

我们使用 [GitHub Issues](https://github.com/remaxjs/remax/issues) 来管理 bug。 如果你想要你发现的 bug 被快速解决，最好的办法就是给我们提一个带有可重现仓库的 issue。

所有的 bug 会有 [bug 标记](https://github.com/remaxjs/remax/issues?q=is%3Aissue+is%3Aopen+label%3Abug)，如果你想找一个容易上手的 bug 来修复，可以查看标记为 [good first issue](https://github.com/remaxjs/remax/issues?q=is%3Aissue+is%3Aclosed+label%3A%22good+first+issue%22) 的 issue。

在你报告一个 bug 之前，请先确保已经搜索过已有的 issues。

## 开发

在开始之前推荐你先了解一下[《Remax 实现原理》](https://remaxjs.org/guide/implementation-notes)

### 构建项目

> 推荐用 yarn

```bash
$ git clone git@github.com:remaxjs/remax.git
$ cd remax
$ yarn
$ yarn watch
```

### 调试开发包

```bash
$ cd path/to/remax/packages/remax
$ yarn link
$ cd path/to/remax/packages/@remax/cli
$ yarn link
$ git clone git@github.com:remaxjs/examples.git remax-examples
$ cd remax-examples/ali # 找一个你要调试的例子
$ yarn link remax @remax/cli
$ yarn dev
```

然后用对应小程序的 IDE 打开 `dist` 目录就可以调试了。

### 运行测试

```javascript
$ yarn test
```
