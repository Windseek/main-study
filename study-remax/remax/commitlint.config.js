module.exports = {
  extends: ['@commitlint/config-conventional'],
  rules: {
    'scope-enum': [2, 'always', ['cli', 'ali', 'wechat', 'remax', 'toutiao', 'one', 'web']],
    'type-enum': [
      2,
      'always',
      ['build', 'ci', 'chore', 'docs', 'feat', 'fix', 'perf', 'refactor', 'revert', 'style', 'test', 'deps'],
    ],
  },
};
