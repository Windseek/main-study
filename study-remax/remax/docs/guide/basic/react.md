---
title: React
order: 21
---

# React

在使用 Remax 前你需要对 [React](https://reactjs.org/) 有基本的了解。

如果你有 React 的开发经验，那么也要注意 Remax 不同于 React DOM。Remax 使用小程序提供的组件来构建应用，而且你的代码是运行在小程序环境中。

你在 Remax 中使用 React 没有任何特殊限制，同时也支持 ES6 语法， ES Module，async/await 等等。
