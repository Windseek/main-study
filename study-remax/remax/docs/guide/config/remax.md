---
title: Remax 配置
order: 1
group:
  title: 配置
  order: 3
---

# Remax 配置

通过 `remax.config.js`  修改 Remax 配置。

```js
// remax.config.js
module.exports = {
  output: 'build',
};
```

## `cwd`

配置项目路径，默认当前路径。

_默认值：_`process.cwd()`。

## `rootDir`

指定代码的根目录。

_默认值：_`'src'`。

## `ouput`

指定输出目录。

_默认值：_`'dist'`,

## `pxToRpx`

是否将 `px` 转换为 `rpx`。

_默认值：_ `true`

## `configWebpack`

修改 webpack 的配置。

```javascript
// remax.config.js

module.exports = {
  configWebpack({ config, webpack, addCSSRule }) {
    // config 是的 https://github.com/neutrinojs/webpack-chain Config 对象。
    config.plugins.delete('webpackbar'); // 去掉进度条
    config.plugin('custom-define').use(webpack.DefinePlugin, [
      {
        __MOCK__: JSON.stringify(process.env.MOCK),
      },
    ]); // 新增一个 define 插件

    // 引用 wxss
    addCSSRule({
      name: 'wxss',
      test: /\.wxss(\?.*)?$/,
    });
  },
};
```

*默认值：*无

## `turboPages`

开启优化编译的页面，具体可以参考[《性能优化》](/guide/advanced/performance)。

_默认值：_`[]`

## `compressTemplate`

是否对生成的模板文件进行压缩，如果在预览小程序时碰到体积超限，可以尝试开启这个选项。

_默认值：_`process.env.NODE_ENV === 'production'`
