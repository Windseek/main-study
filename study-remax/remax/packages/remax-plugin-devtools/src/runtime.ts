import { connectToDevTools } from '@remax/react-devtools-core';
import WebSocket from './WebSocket';

export default {
  onAppConfig({ config }: { config: any }) {
    const originalLaunch = config.onLaunch;
    config.onLaunch = function (...args: any) {
      try {
        console.log('尝试连接 React DevTools，请忽略连接错误信息，详情请参考 https://remaxjs.org/guide/basic/devtools');
        connectToDevTools({
          websocket: new WebSocket('ws://127.0.0.1:8097'),
        });
      } catch (e) {
        // ignore
      }
      originalLaunch.apply(config, args);
    };

    return config;
  },
};
