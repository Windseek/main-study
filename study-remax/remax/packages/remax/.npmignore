node_modules/
src/
__tests__/
yarn-error.log
.DS_Store
coverage/
