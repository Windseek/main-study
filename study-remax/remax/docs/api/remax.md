---
title: remax
order: 0
nav:
  title: API
  order: 1
---

_带有 `?` 的选项表示非必填项_

## useQuery()

获取页面参数 hook

```jsx
import { useQuery } from 'remax';

export default () => {
  const query = useQuery();

  ...
}
```

#### 参数

无

#### 返回值

返回一个页面参数键值对象。

## usePageInstance()

获取当前页面组件实例

```js
import { usePageInstance } from 'remax';

export default () => {
  const instance= usePageInstance();

  ...
}
```

#### 参数

无

#### 返回值

返回当前页面实例

## useNativeEffect(callback, deps?)

原生 setData 回调执行 hook。用于在 React rerender 后，小程序真正更新的时机的 hook。
在 Remax 的机制中，`React.useEffect` 只能反映 React 的 render 完成了，不代表小程序渲染完成。因此通过 `useNativeEffect` 来处理小程序 setData 的 callback 回调。

```jsx
import * as React from 'react';
import { useNativeEffect } from 'remax';

export default () => {
  const [width, setWidth] = React.useState();

  React.useEffect(() => {
    console.log('width 更新在 React 中生效了');
  }, [width])

  useNativeEffect(() => {
    console.log('width 更新在小程序中真正生效了');
}, [width])

  ...
}
```

#### 参数

1. `callback` 回调函数，与 `useEffect` 类似，可以返回一个函数处理 unsubscribe 时的逻辑。
2. `deps` 依赖参数，与 `useEffect` 类似，只有依赖项发生了变化，才会触发回调执行。

#### 返回值

无
