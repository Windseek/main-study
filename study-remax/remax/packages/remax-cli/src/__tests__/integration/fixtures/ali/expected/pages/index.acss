@import "./../remax-styles.acss";
.b {
  background-color: white;
}

.page-index___1ZW9Y {
  background-color: aliceblue;
}

