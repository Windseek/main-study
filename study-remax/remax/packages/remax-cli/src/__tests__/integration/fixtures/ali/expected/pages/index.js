require('./../runtime.js');
require('./../remax-vendors.js');
require('./../remax-styles.js');
(my["webpackJsonp"] = my["webpackJsonp"] || []).push([[2],[
/* 0 */,
/* 1 */,
/* 2 */,
/* 3 */,
/* 4 */,
/* 5 */,
/* 6 */
/***/ (function(module, exports) {

module.exports = require("@remax/runtime");

/***/ }),
/* 7 */,
/* 8 */
/***/ (function(module, exports) {

module.exports = require("react");

/***/ }),
/* 9 */,
/* 10 */
/***/ (function(module, exports, __webpack_require__) {

module.exports = __webpack_require__(11);


/***/ }),
/* 11 */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _remax_runtime__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(6);
/* harmony import */ var _remax_runtime__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(_remax_runtime__WEBPACK_IMPORTED_MODULE_0__);
/* harmony import */ var _index_js__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(13);
__webpack_require__(12);



Page(Object(_remax_runtime__WEBPACK_IMPORTED_MODULE_0__["createPageConfig"])(_index_js__WEBPACK_IMPORTED_MODULE_1__["default"], 'pages/index'));

/***/ }),
/* 12 */
/***/ (function(module, exports, __webpack_require__) {

// extracted by mini-css-extract-plugin

/***/ }),
/* 13 */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var regenerator_runtime__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(14);
/* harmony import */ var regenerator_runtime__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(regenerator_runtime__WEBPACK_IMPORTED_MODULE_0__);
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(8);
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_1___default = /*#__PURE__*/__webpack_require__.n(react__WEBPACK_IMPORTED_MODULE_1__);
/* harmony import */ var remax__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(15);
/* harmony import */ var remax__WEBPACK_IMPORTED_MODULE_2___default = /*#__PURE__*/__webpack_require__.n(remax__WEBPACK_IMPORTED_MODULE_2__);
/* harmony import */ var _components_C__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(16);
/* harmony import */ var _data_json__WEBPACK_IMPORTED_MODULE_4__ = __webpack_require__(18);
var _data_json__WEBPACK_IMPORTED_MODULE_4___namespace = /*#__PURE__*/__webpack_require__.t(18, 1);
/* harmony import */ var _index_module_css_modules__WEBPACK_IMPORTED_MODULE_5__ = __webpack_require__(19);
/* harmony import */ var _index_module_css_modules__WEBPACK_IMPORTED_MODULE_5___default = /*#__PURE__*/__webpack_require__.n(_index_module_css_modules__WEBPACK_IMPORTED_MODULE_5__);


function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }



 // 测试引用 json 模块



/* harmony default export */ __webpack_exports__["default"] = (function () {
  var _;

  var props = {};
  var TextElement = /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_1__["cloneElement"]( /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_1__["createElement"](remax__WEBPACK_IMPORTED_MODULE_2__["Text"], null));

  function handleClick() {
    return _handleClick.apply(this, arguments);
  }

  function _handleClick() {
    _handleClick = _asyncToGenerator( /*#__PURE__*/regenerator_runtime__WEBPACK_IMPORTED_MODULE_0___default.a.mark(function _callee() {
      return regenerator_runtime__WEBPACK_IMPORTED_MODULE_0___default.a.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _context.next = 2;
              return Promise.resolve(1);

            case 2:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return _handleClick.apply(this, arguments);
  }

  function handleTouchStart() {}

  var value = (_ = 0) !== null && _ !== void 0 ? _ : 1;
  return /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_1__["createElement"](remax__WEBPACK_IMPORTED_MODULE_2__["View"], {
    className: _index_module_css_modules__WEBPACK_IMPORTED_MODULE_5___default.a['page-index']
  }, /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_1__["createElement"](_components_C__WEBPACK_IMPORTED_MODULE_3__["default"], {
    className: "b"
  }), /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_1__["createElement"](remax__WEBPACK_IMPORTED_MODULE_2__["View"], _extends({
    onClick: handleClick,
    onTouchStart: handleTouchStart,
    id: "view",
    "unrecognized-props": "",
    "data-foo": "bar",
    key: "my-view"
  }, props, {
    __key: "my-view"
  }), "foo", _data_json__WEBPACK_IMPORTED_MODULE_4__ === null || _data_json__WEBPACK_IMPORTED_MODULE_4__ === void 0 ? void 0 : _data_json__WEBPACK_IMPORTED_MODULE_4__.data, value), TextElement);
});

/***/ }),
/* 14 */
/***/ (function(module, exports) {

module.exports = require("regenerator-runtime");

/***/ }),
/* 15 */
/***/ (function(module, exports) {

module.exports = require("remax");

/***/ }),
/* 16 */,
/* 17 */,
/* 18 */
/***/ (function(module) {

module.exports = JSON.parse("{\"data\":\"value\"}");

/***/ }),
/* 19 */
/***/ (function(module, exports, __webpack_require__) {

// extracted by mini-css-extract-plugin
module.exports = {"page-index":"page-index___1ZW9Y"};

/***/ })
],[[10,0,6,4]]]);