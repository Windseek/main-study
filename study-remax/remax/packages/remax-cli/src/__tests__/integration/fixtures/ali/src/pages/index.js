import * as React from 'react';
import { View, Text } from 'remax';
import C from '@/components/C';
// 测试引用 json 模块
import dataJson from './data.json';
import styles from './index.module.css';

export default () => {
  const props = {};
  const TextElement = React.cloneElement(<Text />);

  async function handleClick() {
    await Promise.resolve(1);
  }

  function handleTouchStart() {}
  const value = 0 ?? 1;

  return (
    <View className={styles['page-index']}>
      <C className="b" />
      <View
        onClick={handleClick}
        onTouchStart={handleTouchStart}
        id="view"
        unrecognized-props=""
        data-foo="bar"
        key="my-view"
        {...props}
      >
        foo
        {dataJson?.data}
        {value}
      </View>
      {TextElement}
    </View>
  );
};
