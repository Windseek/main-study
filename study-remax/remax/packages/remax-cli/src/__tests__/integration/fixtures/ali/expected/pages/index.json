{
  "defaultTitle": "Ali App",
  "titleBarColor": "#323239",
  "usingComponents": {}
}